packages/react/src/metadata.tsx

```diff
@@ -408,6 +447,12 @@ const specialModelKeys = new Set(["DataModel-Shopify-Shop"]);
 const isAcceptedFieldType = (field: FieldMetadata) => acceptedAutoFormFieldTypes.has(field.fieldType);
 const isAcceptedUpsertFieldType = (field: FieldMetadata) => field.fieldType === FieldType.Id || isAcceptedFieldType(field);
 
+const isNotJoinModelHasManyField = (field: FieldMetadata) =>
+  field.fieldType !== FieldType.HasMany ||
```

Comment:
I think this would be more clearly expressed in the positive -- isJoinModelHasManyField because some of the double negatives confused me for a second. 

```
(field.fieldType === FieldType.HasMany &&
    field.configuration.__typename === "GadgetHasManyConfig" &&
    field.configuration.isJoinModelHasManyField)
 ```
 
 ...
 
 ```
  // Remove `hasMany` fields that emerge from `hasManyThrough` fields that are not actually model fields
  subset = subset.filter(field => !isJoinModelHasManyField(field));
 ```

Reply:
Updated such that the filter works as a positive check